mobile/lib/widgets/activities/activity_tile.dart

```diff
@@ -14,36 +19,71 @@ class ActivityTile extends HookConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
+    Future<void> onTap() async {
+      if (activity.assetId == null) {
+        return;
+      }
+
+      final asset = await ref.read(assetRepositoryProvider).getByRemoteId(
+            activity.assetId!,
+          );
+      if (asset == null) {
+        return;
+      }
+
+      final renderList = await RenderList.fromAssets(
+        [asset],
+        GroupAssetsBy.none,
+      );
+      final assetNotifier = ref.read(currentAssetProvider.notifier);
+      assetNotifier.set(asset);
```

Comment:
Why are you setting the current asset to null after pushing the route? Doing the following before pushing the route should be enough. We don't have to do anything after
```suggestion
      ref.read(currentAssetProvider.notifier).set(asset);
```

Reply:
There's a reason actually. Activity page opened from GalleryViewer and activity page in album is watching the same notifier (obviously, since it's the same page). Without setting current asset to null on return from gallery activity page will list comments/likes only for asset that was set in 

```dart
ref.read(currentAssetProvider.notifier).set(asset)
```

The other approach would be to pass activity page "mode" (album activities or single asset activities) in page navigation arguments. Then, this code can be omitted, as you suggested.

Reply:
Good catch!